ArduPlane/quadplane.cpp

```diff
@@ -1283,6 +1298,47 @@ float QuadPlane::desired_auto_yaw_rate_cds(void) const
     return yaw_rate;
 }
 
+/*
+  start altitude assistance
+*/
+void QuadPlane::start_alt_assist(float height_above_ground)
+{
+    assist_alt.in_assist = true;
+    gcs().send_text(MAV_SEVERITY_INFO, "Alt assist %.1fm", height_above_ground);
+    switch (AltAssistAction(assist_alt.action.get())) {
+    case AltAssistAction::None:
+        break;
+    case AltAssistAction::QHover:
+        if (plane.rc_failsafe_active()) {
+            plane.set_mode(plane.mode_qland, ModeReason::ALT_ASSIST);
+        } else {
+            plane.set_mode(plane.mode_qhover, ModeReason::ALT_ASSIST);
+        }
+        break;
+    case AltAssistAction::QLoiter:
+        if (plane.rc_failsafe_active()) {
+            plane.set_mode(plane.mode_qland, ModeReason::ALT_ASSIST);
+        } else {
+            plane.set_mode(plane.mode_qloiter, ModeReason::ALT_ASSIST);
+        }
+        break;
+    case AltAssistAction::QLand:
+        plane.set_mode(plane.mode_qland, ModeReason::ALT_ASSIST);
+        break;
+    case AltAssistAction::QRTL:
+        plane.set_mode(plane.mode_qrtl, ModeReason::ALT_ASSIST);
```

Comment:
I wonder if we need to make sure this is actually a QRTL, and does not switch to forward flight and do an approach. My understanding is that if this assist alt ever trips is suggests that the vehicle may not be capable of forward flight at all.